iot/end_to_end_example/device/app/build.gradle

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+
+android {
+    compileSdkVersion 27
+    buildToolsVersion '27.0.3'
+
+    defaultConfig {
+        applicationId "com.example.androidthings.coffeecontrol"
+        minSdkVersion 27
+        targetSdkVersion 27
+        versionCode 1
+        versionName "1.0"
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    buildTypes {
+        debug {
+            buildConfigField "String", "PROJECT_ID", '"YOUR_PROJECT_ID"'
+            buildConfigField "String", "REGISTRY_ID", '"YOUR_REGISTRY_ID"'
+            buildConfigField "String", "DEVICE_ID", '"YOUR_DEVICE_ID"'
+            buildConfigField "String", "ZONE", '"us-central1"'
+        }
+        release {
+            initWith(buildTypes.debug)
+
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+
+
+dependencies {
+    testImplementation 'junit:junit:4.12'
+
+    compileOnly 'com.google.android.things:androidthings:1.0'
+
+    implementation 'com.google.android.things:cloud-iot-core:1.0.0'
+    implementation 'com.google.android.things.contrib:driver-button:1.0'
+    implementation 'com.google.android.things.contrib:driver-ht16k33:1.0'
+    implementation 'com.google.android.things.contrib:driver-apa102:1.0'
+
+    implementation('com.google.api-client:google-api-client-android:1.22.0') {
+        exclude group: 'org.apache.httpcomponents'
+    }
+    implementation('com.google.apis:google-api-services-pubsub:v1-rev12-1.22.0') {
+        exclude group: 'org.apache.httpcomponents'
+    }
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "com.beust:klaxon:3.0.4"
+}
+repositories {
```

Comment:
I believe this can be removed. You are already using `google()` and `jcenter()`.